src/explore-education-statistics-frontend/src/modules/find-statistics/FindStatisticsPageNew.tsx

```diff
@@ -2,71 +2,108 @@ import Button from '@common/components/Button';
 import SearchForm from '@frontend/modules/find-statistics/components/SearchForm';
 import SortControls from '@frontend/modules/find-statistics/components/SortControls';
+import createPublicationListRequest from '@frontend/modules/find-statistics/utils/createPublicationListRequest';
 import { logEvent } from '@frontend/services/googleAnalyticsService';
+import compact from 'lodash/compact';
+import isEqual from 'lodash/isEqual';
 import omit from 'lodash/omit';
 import { NextPage } from 'next';
 import { useRouter } from 'next/router';
 import { ParsedUrlQuery } from 'querystring';
-import React from 'react';
-
-interface FindStatisticsPageQuery {
-  page?: number;
-  releaseType?: ReleaseType;
-  search?: string;
-  sortBy?: PublicationSortOption;
-  themeId?: string;
-}
+import React, { useState, useEffect, useRef } from 'react';
 
 interface Props {
   paging: Paging;
   publications: PublicationListSummary[];
-  searchTerm?: string;
-  sortBy?: PublicationSortOption;
+  query: FindStatisticsPageQuery;
+  themes: ThemeSummary[];
 }
 
 const FindStatisticsPageNew: NextPage<Props> = ({
-  paging,
-  publications,
-  searchTerm,
-  sortBy = 'newest',
+  paging: initialPaging,
+  publications: initialPublications,
+  query,
+  themes,
 }) => {
-  const { page, totalPages, totalResults } = paging;
   const router = useRouter();
   const isLoading = useRouterLoading();
+
+  const [currentQuery, setQuery] = useState<FindStatisticsPageQuery>(query);
+  const { releaseType, search, sortBy = 'newest', themeId } = currentQuery;
+
+  const [publications, setPublications] = useState<PublicationListSummary[]>(
+    initialPublications,
+  );
+  const [paging, setPaging] = useState<Paging>(initialPaging);
+  const { page, totalPages, totalResults } = paging;
+
   const { isMedia: isMobileMedia } = useMobileMedia();
+  const mobileFilterButtonRef = useRef<HTMLButtonElement>(null);
+  const [showMobileFilters, toggleMobileFilters] = useToggle(false);
+
+  const isFiltered = !!search || !!releaseType || !!themeId;
+
+  const selectedTheme = themes.find(theme => theme.id === themeId);
+  const selectedReleaseType = releaseTypes[releaseType as ReleaseType];
 
-  // TODO EES-3517 - include filtering by theme etc here;
-  const isFiltered = !!searchTerm;
+  const filteredByString = compact([
+    search,
+    selectedTheme?.title,
+    selectedReleaseType,
+  ]).join(', ');
 
-  const updateQueryParams = async (query: FindStatisticsPageQuery) => {
+  useEffect(() => {
+    async function fetchPublications(nextQuery: FindStatisticsPageQuery) {
+      const updatedPublications = await publicationService.listPublications(
+        nextQuery,
+      );
+      setPublications(updatedPublications.results);
+      setPaging(updatedPublications.paging);
+    }
+    if (
+      Object.keys(router.query).length &&
+      !isEqual(router.query, currentQuery)
+    ) {
+      setQuery(router.query);
+      fetchPublications(createPublicationListRequest(router.query));
+    }
```

Comment:
@ntsim I did think about using react-query originally and thought it might be overkill here, but have given it a go and it works well. 
I ran out of time today so haven't got very far with the tests, I'll pick that back up on Tuesday. If you can review the changes for react-query in the meantime that'd be good, I've kept it in a separate commit for now so it's clear what's changed.